quantities/src/main/java/com/decathlon/vitamin/compose/quantities/VitaminQuantities.kt

```diff
@@ -0,0 +1,299 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Divider
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.decathlon.vitamin.compose.foundation.VitaminTheme
+import com.decathlon.vitamin.compose.foundation.VtmnStatesDisabled
+
+private const val VtmnStatesEnabled = 1.0f
+
+object VitaminQuantities {
+
+    @Composable
+    fun Primary(
+        value: String,
+        modifier: Modifier = Modifier,
+        addEnabled: Boolean = true,
+        subtractEnabled: Boolean = true,
+        editTextEnabled: Boolean = true,
+        hasQuantityError: Boolean = false,
+        keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+        colors: QuantityColors = VitaminQuantitiesColors.primary(),
+        onAddClicked: () -> Unit,
+        onSubtractClicked: () -> Unit,
+        onValueChange: (String?) -> Unit,
+    ) {
+        Row(
+            modifier = modifier
+                .height(48.dp)
+                .width(154.dp)
+        ) {
+            LeftButton(onSubtractClicked, colors, subtractEnabled)
+
+            MiddleTextField(
+                modifier = Modifier
+                    .background(
+                        color = colors.backgroundColor.copy(
+                            alpha = if (!addEnabled && !subtractEnabled) VtmnStatesDisabled else VtmnStatesEnabled
+                        )
+                    )
+                    .fillMaxHeight()
+                    .weight(1.0f),
+                addEnabled = addEnabled,
+                subtractEnabled = subtractEnabled,
+                hasQuantityError = hasQuantityError,
+                colors = colors,
+                value = value,
+                onValueChange = onValueChange,
+                editTextEnabled = editTextEnabled,
+                keyboardOptions = keyboardOptions
+
+            )
+
+            RightButton(onAddClicked = onAddClicked, addEnabled = addEnabled, colors = colors)
+        }
+    }
+}
+
+@Composable
+private fun MiddleTextField(
+    modifier: Modifier,
+    addEnabled: Boolean,
+    subtractEnabled: Boolean,
+    hasQuantityError: Boolean,
+    colors: QuantityColors,
+    value: String,
+    onValueChange: (String?) -> Unit,
```

Comment:
Since `value` is a quantity and since we have, by default, `KeyboardOptions(keyboardType = KeyboardType.Number)`, shouldn't we have `onValueChange: (String) -> Unit` instead?
I don't see how the User can provide a `null` value.